examples/controlnet/train_controlnet_flux.py

```diff
@@ -0,0 +1,1321 @@
+        latents = latents.view(batch_size, num_channels_latents, height // 2, 2, width // 2, 2)
+        latents = latents.permute(0, 2, 4, 1, 3, 5)
+        latents = latents.reshape(batch_size, (height // 2) * (width // 2), num_channels_latents * 4)
+
+        return latents
+
+    image_logs = None
+    for epoch in range(first_epoch, args.num_train_epochs):
+        for step, batch in enumerate(train_dataloader):
+            with accelerator.accumulate(flux_controlnet):
+                # Convert images to latent space
+
+                # vae encode
+                pixel_values = batch["pixel_values"].to(dtype=weight_dtype)
+                pixel_latents_tmp = vae.encode(pixel_values).latent_dist.sample()
+                pixel_latents_tmp = (pixel_latents_tmp - vae.config.shift_factor) * vae.config.scaling_factor
+                pixel_latents = _pack_latents(
+                    pixel_latents_tmp,
+                    pixel_values.shape[0],
+                    pixel_latents_tmp.shape[1],
+                    pixel_latents_tmp.shape[2],
+                    pixel_latents_tmp.shape[3],
+                )
+
+                control_values = batch["conditioning_pixel_values"].to(dtype=weight_dtype)
+                control_latents = vae.encode(control_values).latent_dist.sample()
+                control_latents = (control_latents - vae.config.shift_factor) * vae.config.scaling_factor
+                control_image = _pack_latents(
+                    control_latents,
+                    control_values.shape[0],
+                    control_latents.shape[1],
+                    control_latents.shape[2],
+                    control_latents.shape[3],
+                )
+
+                latent_image_ids = _prepare_latent_image_ids(
+                    batch_size=pixel_latents_tmp.shape[0],
+                    height=pixel_latents_tmp.shape[2],
+                    width=pixel_latents_tmp.shape[3],
+                    device=pixel_values.device,
+                    dtype=pixel_values.dtype,
+                )
+
+                # Sample noise that we'll add to the latents
+                noise = torch.randn_like(pixel_latents).to(accelerator.device).to(dtype=weight_dtype)
+                bsz = pixel_latents.shape[0]
+
+                # Sample a random timestep for each image
+                t = torch.sigmoid(torch.randn((bsz,), device=accelerator.device, dtype=weight_dtype))
+
+                # apply flow matching
+                noisy_latents = (
+                    1 - t.unsqueeze(1).unsqueeze(2).repeat(1, pixel_latents.shape[1], pixel_latents.shape[2])
+                ) * pixel_latents + t.unsqueeze(1).unsqueeze(2).repeat(
+                    1, pixel_latents.shape[1], pixel_latents.shape[2]
+                ) * noise
+
+                guidance_vec = torch.full(
+                    (noisy_latents.shape[0],), 3.5, device=noisy_latents.device, dtype=weight_dtype
```

Comment:
Let's make `3.5` a CLI argument `guidance_scale`.

Reply:
added in this [commit](https://github.com/huggingface/diffusers/pull/9324/commits/32fbeac2bfd7a5434b6c05b581dff608730b56e3) Plz take a look at it.